### ISSUE TITLE
Unify dropdown functionality across all dropdowns

### ISSUE THREAD

**Issue by hom3mad3** [liqd/a4-meinberlin#5587]
## **Issue:**
`dropdown.js` lacks certain features found in `user_indicator.js`, such as keyboard navigation and close-on-click-outside functionality. However, `user_indicator.js` isn't easily modularized due to its design, which focuses on a single element without considering component reusability.

## **Proposal:**
To address this, I'd suggest refactoring `user_indicator.js` onto `dropdown.js` to integrate the missing features, enhancing the overall reusability of dropdown functionality. This will ensure that all dropdowns use one JS file, even when multiple instances are present on the page.

## 1. **dropdown.js**
element triggered with `dropdown.js` and styled by `scss/components_user_facing/_dropdown.scss`:

<img src="https://github.com/liqd/a4-meinberlin/assets/8156337/c351d568-c9fe-4611-a1e8-283f341dc80d" alt="item-detail-dropdown" width="300px">

## 2. **user_indicator.js**
event is triggered w/ `user_indicator.js` and styled by `/scss/components_user_facing/_user_indicator.scss`:

<img src="https://github.com/liqd/a4-meinberlin/assets/8156337/bf7ce748-717a-4e03-97db-f2aa167243ef" alt="user-dropdown" width="300px">
